Sources/AsyncConnection.swift

```diff
@@ -3,7 +3,7 @@ public protocol AsyncConnection: AsyncStream {
 }
 
 extension AsyncConnection {
-    public func open(result: (Void throws -> AsyncConnection) -> Void) throws {
+    public func open(_ result: (Void throws -> AsyncConnection) -> Void) throws {
```

Comment:
> I think we shouldn’t have typealiases in the library of standards

That's fair. They make it much more difficult to reason about what is happening at first glance because you don't know what the typealias equals.

Also `completion` is not a perfect name. It's a closure that is handed a result closure. This seems like a more specific type of completion callback.